src/WLR/Types/InputDevice.hsc

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE EmptyDataDeriving #-}
+
+module WLR.Types.InputDevice where
+
+#define WLR_USE_UNSTABLE
+#include <wlr/types/wlr_input_device.h>
+
+import Foreign
+import Foreign.C.Types
+
+import WL.ServerCore
+
+type WLR_input_device_type = CInt
+#enum WLR_input_device_type, , \
+    WLR_INPUT_DEVICE_KEYBOARD, \
+    WLR_INPUT_DEVICE_POINTER, \
+    WLR_INPUT_DEVICE_TOUCH, \
+    WLR_INPUT_DEVICE_TABLET_TOOL, \
+    WLR_INPUT_DEVICE_TABLET_PAD, \
+    WLR_INPUT_DEVICE_SWITCH
+
+data {-# CTYPE "wlr/types/wlr_input_device.h" "struct wlr_input_device" #-} WLR_input_device
+    = WLR_input_device
+    { wlr_input_device_type :: WLR_input_device_type
+    , wlr_input_device_vendor :: CUInt
+    , wlr_input_device_product :: CUInt
+    , wlr_input_device_name :: Ptr CChar
```

Comment:
This could be declared as a `CString`, though double-checking that’s just a synonym for `Ptr CChar` anyway.